autoload/neomake/signs.vim

```diff
@@ -125,8 +125,15 @@ function! neomake#signs#PlaceVisibleSigns() abort
         if !has_key(s:sign_queue[type], buf)
             continue
         endif
-        let topline = line('w0')
-        let botline = line('w$')
+
+        if !get(g:, 'neomake_place_signs_at_once', 0)
+            let topline = line('w0')
+            let botline = line('w$')
+        else
+            let topline = line('0')
+            let botline = line('$')
```

Comment:
I agree that this should be the default behaviour. I didn't feel confident enough to change it, so I've just added an option which solves the problem for me. 

In general, I think that the best solution would be doing something like [vim-gitgutter](https://github.com/airblade/vim-gitgutter) is doing, which is always placing all the signs at once, but have a configurable limit of the number of signs to be placed, to prevent vim from being unresponsive if there are too many signs.